components/director/internal/authenticator/middleware_test.go

```diff
@@ -0,0 +1,277 @@
+
+		rr := httptest.NewRecorder()
+
+		req, err := http.NewRequest("GET", "/", nil)
+		require.NoError(t, err)
+
+		req.Header.Add("tenant", tnt)
+
+		token, err := createToken(tnt, scopes)
+		require.NoError(t, err)
+		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+
+		//when
+		middleware(handler).ServeHTTP(rr, req)
+		//then
+		assert.Equal(t, "while parsing token: Unexpected signing method: none\n", rr.Body.String())
+		assert.Equal(t, http.StatusUnauthorized, rr.Code)
+
+	})
+
+	t.Run("Error - tenant header is not provided", func(t *testing.T) {
+		//given
+		middleware := createMiddleware(t, false)
+		handler := testHandler(t, scopes)
+
+		rr := httptest.NewRecorder()
+
+		req, err := http.NewRequest("GET", "/", nil)
+		require.NoError(t, err)
+
+		//when
+		middleware(handler).ServeHTTP(rr, req)
+
+		//then
+		assert.Equal(t, rr.Code, 400)
+		assert.Equal(t, "No tenant header\n", rr.Body.String())
+	})
+
+	t.Run("Error - can't get token from header", func(t *testing.T) {
+		//given
+		middleware := createMiddleware(t, false)
+		handler := testHandler(t, scopes)
+		rr := httptest.NewRecorder()
+
+		req, err := http.NewRequest("GET", "/", nil)
+		require.NoError(t, err)
+
+		req.Header.Add("tenant", tnt)
+
+		//when
+		middleware(handler).ServeHTTP(rr, req)
+
+		//then
+		assert.Equal(t, "Invalid bearer token\n", rr.Body.String())
+		assert.Equal(t, 400, rr.Code)
+	})
+
+	t.Run("Error - can't parse token", func(t *testing.T) {
+		//given
+		middleware := createMiddleware(t, false)
+		handler := testHandler(t, scopes)
+		rr := httptest.NewRecorder()
+
+		req, err := http.NewRequest("GET", "/", nil)
+		require.NoError(t, err)
+
+		req.Header.Add("tenant", tnt)
+		req.Header.Add("Authorization", "Bearer fake-token")
+
+		//when
+		middleware(handler).ServeHTTP(rr, req)
+
+		//then
+		assert.Equal(t, "while parsing token: token contains an invalid number of segments\n", rr.Body.String())
+		assert.Equal(t, http.StatusUnauthorized, rr.Code)
+	})
+
+	t.Run("Error - Token signed with different key", func(t *testing.T) {
+		//given
+		middleware := createMiddleware(t, false)
+		handler := testHandler(t, scopes)
+
+		privateJWKS2, err := authenticator.FetchJWK(PrivateJWKS2URL)
+		require.NoError(t, err)
+
+		rr := httptest.NewRecorder()
+
+		req, err := http.NewRequest("GET", "/", nil)
+		require.NoError(t, err)
+
+		req.Header.Add("tenant", tnt)
+
+		token, err := createTokenWithSigningMethod(t, tnt, scopes, privateJWKS2.Keys[0])
+		require.NoError(t, err)
+		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+
+		//when
+		middleware(handler).ServeHTTP(rr, req)
+
+		//then
+		assert.Equal(t, http.StatusUnauthorized, rr.Code)
+		assert.Equal(t, "while parsing token: crypto/rsa: verification error\n", rr.Body.String())
+	})
+
+}
+
+type jwtTokenClaims struct {
+	Scopes []string `json:"scopes"`
+	Tenant string   `json:"tenant"`
+	jwt.StandardClaims
+}
+
+func createToken(t string, scopes []string) (string, error) {
```

Comment:
1. what is `t` - usually it is `testing.T`
2. in other places you have testing.T and internally validate errors, please make it consistent